releasedir/provider.go

```diff
@@ -103,6 +103,8 @@ func (p Provider) newBlobstore(dirPath string) boshblob.DigestBlobstore {
 		blobstore = boshblob.NewLocalBlobstore(p.fs, p.uuidGen, options)
 	case "s3":
 		blobstore = NewS3Blobstore(p.fs, p.uuidGen, options)
```

Comment:
There seems to be a lot of duplication between NewGCSBlobstore/NewS3Blobstore GCSBlobstore/S3Blobstore. Did you consider refactoring `S3Blobstore` into something that accepts a `client` object (instead of creating one)?

I'd only go that route if everything but client creation is the same, but I think it is?